enterprise/backend/test/metabase_enterprise/task/cache_test.clj

```diff
@@ -0,0 +1,268 @@
+(ns metabase-enterprise.task.cache-test
+  (:require
+   [clojure.test :refer :all]
+   [java-time.api :as t]
+   [metabase-enterprise.task.cache :as task.cache]
+   [metabase.public-settings.premium-features :as premium-features]
+   [metabase.query-processor :as qp]
+   [metabase.query-processor.card :as qp.card]
+   [metabase.query-processor.dashboard :as qp.dashboard]
+   [metabase.query-processor.util :as qp.util]
+   [metabase.test :as mt]
+   [metabase.test.fixtures :as fixtures]
+   [toucan2.core :as t2]))
+
+(use-fixtures :once (fixtures/initialize :db))
+
+(set! *warn-on-reflection* true)
+
+(defn parameterized-native-query
+  "A native query with an optional template tag"
+  []
+  {:database (mt/id)
+   :type     :native
+   :native   {:template-tags {"date" {:name         "date"
+                                      :display-name "Check-In Date"
+                                      :type         :text}}
+              :query "SELECT count(*) FROM CHECKINS [[WHERE date = {{date}}]]"}})
+
+(defn run-query-for-card-id
+  [card-id params]
+  (mt/as-admin
+    (qp.card/process-query-for-card
+     card-id :api
+     :parameters params
+     :make-run (constantly
+                (fn [query info]
+                  (qp/process-query (qp/userland-query (assoc query :info info))))))))
+
+(defn run-query-for-dashcard
+  [card-id dashboard-id dashcard-id]
+  (mt/as-admin
+    (qp.dashboard/process-query-for-dashcard
+     :card-id card-id
+     :dashboard-id dashboard-id
+     :dashcard-id dashcard-id
+     :make-run (constantly
+                (fn [query info]
+                  (qp/process-query (qp/userland-query (assoc query :info info))))))))
+
+(defn- clear-query-tables!
+  "Wipes the the query, query_execution, and query_cache tables prior to a test run, to ensure we're testing a
+  consistent state. These are all auto-populated tables so clearing them shouldn't impact the functioning of Metabase."
+  []
+  (doall (map t2/delete! [:model/Query :model/QueryExecution :model/QueryCache])))
```

Comment:
I'm not in love with this approach — but it's difficult to test the logic around finding queries to rerun without a guarantee that these tables have consistent state. But this does preclude these tests from ever running in parallel. If anyone has a suggestion of how to make these parallelizable I'd love to hear it.